### ISSUE TITLE
Missing iOS's implementations

### ISSUE THREAD

**Issue by juandausa** [flutter/flutter#111583]
## Use case
The current implementation of the [file_selector plugin](https://github.com/flutter/plugins/tree/main/packages/file_selector) has missing features on iOS.

<img src="https://user-images.githubusercontent.com/7011782/190223623-8569e224-353b-44ea-8627-74da46b1bed7.png"  width="300">
<img src="https://user-images.githubusercontent.com/7011782/190223687-3a35210a-ae19-4154-a1e7-cf85fb7f9713.png"  width="300">

There are [other packages](https://github.com/miguelpruivo/flutter_file_picker) available that offer this implementation that can be used as guidance.

## Proposal
Add the missing implementations to satisfy the whole interface.

**Comment by stuartmorgan** [comment]
One of your pictures is of the test page for `getSavePath`. Can you elaborate on what you expect that method to do on iOS?

**Comment by juandausa** [comment]
Hi Stuart, thanks for your prompt response. About the second image, I would expect that the plugin let users select a directory, like it does on the other platforms.

<img src="https://user-images.githubusercontent.com/7011782/190394871-e6037d70-b0fb-4b21-a75f-600a4e1deae2.png" width="400" />

Here is the code that is being invoked on this page:

```dart
Future<String?> getDirectoryPath({
  String? initialDirectory,
  String? confirmButtonText,
}) async {
  return FileSelectorPlatform.instance.getDirectoryPath(
      initialDirectory: initialDirectory, confirmButtonText: confirmButtonText);
}
```

Thanks,

**Comment by stuartmorgan** [comment]
> About the second image, I would expect that the plugin let users select a directory

Yes, but that's `getDirectoryPath`; I was asking about `getSavePath`.

**Comment by juandausa** [comment]
Hi Stuart, we just want to share the same functionality among all the platforms. We've already verified that the `XFile` abstraction allows developers to store a file in the device.

Thanks,

**Comment by stuartmorgan** [comment]
> we just want to share the same functionality among all the platforms

I understand that, but we're wrapping OS functionality in `file_selector`. What is the iOS version of a save dialog that you are proposing wrapping with `getSavePath`?

> We've already verified that the `XFile` abstraction allows developers to store a file in the device.

`XFile` is an abstraction around files; `file_selector` is an abstraction around native file selector UI. They are very different layers. `getSavePath` doesn't save anything, it shows UI that allows someone to choose where to save something. I'm trying to understand what you are proposing that UI be on iOS.

**Comment by juandausa** [comment]
Hi, please correct me if I'm wrong but, I would propose opening the file selector just to pick a folder, pretty similar to what `getDirectoryPath` will do. 

Thanks,

**Comment by stuartmorgan** [comment]
`getSavePath` returns a file, not a directory, so that would have the wrong semantics.

I'm not aware of any iOS analog to a save dialog, and it sounds like you aren't either, so I'm still not clear on why you believe that the lack of implementation of `getSavePath` is a bug.

**Comment by juandausa** [comment]
Hi Stuart,
Thanks for the clarification, now I have a better understanding of your concern. We explored other packages and found inspiration in this one called [flutter_file_dialog](https://github.com/kineapps/flutter_file_dialog). It allows users to select a folder where the file is going to be saved.

The current code of the flutter_file_dialog doesn't match one-to-one with the file_selector implementation but, it offers a great starting point for developers.

<image src="https://user-images.githubusercontent.com/7011782/191557131-d05599aa-d394-4cca-a7ac-1b20a5f0cfdb.png" width="320px" />


What do you think of offering a UI like this one?

**Comment by stuartmorgan** [comment]
I'm not clear on what that UI is exactly. What component is that displaying?

**Comment by juandausa** [comment]
Hi Stuart,
It is relying on an [UIDocumentPickerViewController](https://github.com/kineapps/flutter_file_dialog/blob/master/ios/Classes/SaveFileDialog.swift).

**Comment by huycozy** [comment]
Hi @juandausa, @stuartmorgan 
I would like to summarize the proposal as below, please correct me if I'm missing something:
- Propose adding implementation to get directory path on iOS: `getDirectoryPath()`
- Proposal adding implementation to save a file on iOS: `getSavePath()`. The proposal UI is relying on an [UIDocumentPickerViewController](https://github.com/kineapps/flutter_file_dialog/blob/master/ios/Classes/SaveFileDialog.swift). For example: https://github.com/flutter/flutter/issues/111583#issuecomment-1253935095

Labeling this issue as a new feature. Thanks!

**Comment by juandausa** [comment]
Hi @huycozy, that's correct. Thanks!
